### PR TITLE
Remove sendgrid and integrate Amazon SES

### LINKED ISSUE #61: Switch to AWS SES
While in general I'm not a fan of overengineering/overcomplicating things, I think we should use [AWS Simple Email Service (SES)](https://aws.amazon.com/ses/) instead of [Sendgrid](https://sendgrid.com/). A couple of reasons why, most of which are logistical and less technical:

* we don't need a separate account
  * this lowers the credential store burden
  * billing becomes easier (see below)
* SES is cheaper (in fact, it's *free*) and easier to bill
  * [according to the SES pricing page](https://aws.amazon.com/ses/pricing/), we get free outbound email for the first 62k emails/mo from an EC2 app; this bot is hosted on EC2
  * in comparison, we use the sendgrid essentials tier (14.95/mo), as the free tier (100/day) is not enough for a very small subset of days (fall gm)
  * in addition, dept will now directly pay for AWS costs instead of requiring reimbursement
* SES is **extremely** stable and has a pretty rich documentation ecosystem

Of course, this does lock us in more to Amazon (ex in the future, switching to Heroku, Azure, GCP, etc. would restart this conversation), but given recent developments with dept I think we'll be with Amazon for a while. That being said, open to discussion & suggestions - I'm sure there's something I'm missing.

I think this is a great feature for someone to tackle over the summer and gain nontrivial ownership over the project and/or become a subject-matter expert of sorts for SES. We **should** wrap this up by September so we can test it for GM.

### PR DESCRIPTION
**Pull request by Saagar3690**
Migrated @sendgrid/mail to Amazon SES.
1. Removed all send grid/mail dependencies
2. Updated config.dev.js and config.prod.js to store Amazon SES credentials
3. New file called sendEmail.js which currently only has a function to send email to users to verify their Discord account in the ACM server, but can add more functions and functionality later as needed
4. Updated main.js and iam.js to use the new sendEmail and remove sgMail

Todo:
- Add credentials for prod in config.prod.js 

Closes #61.

### CODE REVIEW

File: commands/iam.js [hunk 1]
```diff
@@ -13,7 +14,7 @@ function genCode(n) {
 
 // if email has not been verified, send verification code
 // linked to IAM command
-const iam = async function (userid, email, nickname, affiliation, sgMail) {
+const iam = async function (userid, email, nickname, affiliation) {
   // check email against allowed domains
   let domain = email.match(
     '^[a-zA-Z0-9_.+-]+@(?:(?:[a-zA-Z0-9-]+.)?[a-zA-Z]+.)?(' +
```
Comment (advaithg) [inline commands/iam.js:20]: should this be changed to only accept UCLA email IDs?

Comment (Saagar3690) [reply commands/iam.js:20]: I believe Rohit has it already set up so that there's an if to check if it includes the .ucla.edu domain in the email

**Comment by mattxwang** [comment]
Quick note - is the base branch for this right? Looks like we're merging into another branch of yours, not `master`.

**Comment by Saagar3690** [comment]
Yep, Advaith and I agreed to branch it off slash commands because there's some dependency on slash commands, so we wanted to merge it into there while Rohit finishes up reviewing slash commands.

**Review by advaithg — COMMENTED** [review]
looks good as of now, but we should probably wait and make sure it works with the acm credentials before merging it in!

**Pull request merged**